sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -119,18 +125,24 @@ case class AggregateExpression(
       normalizedAggFunc.canonicalized.asInstanceOf[AggregateFunction],
       mode,
       isDistinct,
+      filter,
       ExprId(0))
   }
 
-  override def children: Seq[Expression] = aggregateFunction :: Nil
+  override def children: Seq[Expression] = filter match {
```

Comment:
ditto

Reply:
`override def children: Seq[Expression] = aggregateFunction :: filter.toSeq :: Nil`
This cannot compile.

Reply:
`aggregateFunction +: filter.toSeq`

Reply:
OK